app/assets/javascripts/application.js

```diff
@@ -1,14 +1,54 @@
-//= require rails-ujs
-//= require jquery3
-//= require lodash
-//= require turbolinks
-//= require semantic-ui
-//= require action_cable
-//
-//= require application_init
-//
-// You can generate new channels where WebSocket features live using the `rails generate channel` command.
-//= require_tree ./channels
-//
-//= require common
-//= require_tree ./public
+class Application {
+  static boot() {
+    window.App || (window.App = new Application())
```

Comment:
this seems redundant with admin_init, but maybe they're never loaded together?

Reply:
It is a little redundant, however it'll never be loaded together. But in the event that someone manually does call `.boot()` multiple times, it should only ever create the one instance. I did just move the event handlers into the classes initializer though to make it more idempotentish